app/shared/models/git_repo.rb

```diff
@@ -394,29 +395,38 @@ def setup_auth(repo_auth: self.repo_auth)
         ""
       ].join("\n")
 
-      scope = "local"
+      self.credential_scope = "local"
```

Comment:
Could we do a one-line ternany statement and an `attr_reader` instead of an `attr_accessor` for `credential_scope`?
    
```ruby
@credential_scope = File.directory?(File.join(local_folder, ".git")) ? "local" : "global"
```